docs/annotations.md

```diff
@@ -0,0 +1,286 @@
+## Binding Metadata in Annotations
+
+ During a binding operation, annotations from relevant Kubernetes resources are extracted to gather information about what is interesting for binding. This information is eventually used to bind the application with the backing service by populating the binding Secret.
+
+### Requirements for specifying binding information in a backing service CRD / Kubernetes resource
+
+1. Extract a string from the Kubernetes resource.
+2. Extract a string from the Kubernetes resource, and map it to custom name in the binding Secret.
+3. Extract an entire configmap/Secret from the Kubernetes resource.
+4. Extract a specific field from the configmap/Secret from the Kubernetes resource, and bind it as an environment variable.
+5. Extract a specific field from the configmap/Secret from the Kubernetes resource and and bind it as a volume mount.
+6. Extract a specific field from the configmap/Secret from the Kubernetes resource and map it to different name in the binding Secret.
+7. Extract a “slice of maps” from the Kubernetes resource and generate multiple fields in the binding Secret.
+8. Extract a "slice of strings" from a Kubernetes resource and indicate the content in a specific index in the slice to be relevant for binding.
+
+
+### Data model : Building blocks for expressing binding information
+
+* `path`: A template representation of the path to the element in the Kubernetes resource. The value of `path` could be specified in either [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) or [GO templates](https://golang.org/pkg/text/template/)
+
+* `elementType`: Specifies if the value of the element referenced in `path` is of type `string` / `sliceOfStrings` / `sliceOfMaps`. Defaults to `string` if omitted.
+
+* `objectType`: Specifies if the value of the element indicated in `path` refers to a `ConfigMap`, `Secret` or a plain string in the current namespace!  Defaults to `Secret` if omitted and `elementType` is a non-`string`.
+
+* `bindAs`: Specifies if the element is to be bound as an environment variable or a volume mount using the keywords `envVar` and `volume`, respectively. Defaults to `envVar` if omitted.
```

Comment:
```suggestion
* `bindAs`: Specifies if the element is bound as an environment variable or a volume mount using the keywords `envVar` and `volume`, respectively. Defaults to `envVar` if omitted.
```

Reply:
I did imply "is to be" implying "is supposed to be" or "is expected to be"